src/FakeItEasy/Expressions/ExpressionArgumentConstraintFactory.cs

```diff
@@ -161,9 +161,19 @@ private IArgumentConstraint GetArgumentConstraintFromExpression(Expression expre
 
             object expressionValue = null;
 
-            var constraint = this.argumentConstraintTrapper.TrapConstraintOrCreate(
-                () => expressionValue = expression.Evaluate(),
-                () => CreateEqualityConstraint(expressionValue));
+            IArgumentConstraint constraint;
+            try
+            {
+                constraint = this.argumentConstraintTrapper.TrapConstraintOrCreate(
+                    () => expressionValue = expression.Evaluate(),
+                    () => CreateEqualityConstraint(expressionValue));
+            }
+            catch (TargetInvocationException ex)
+            {
+                throw new UserCallbackException(
+                    ExceptionMessages.UserCallbackThrewAnException($"Argument constraint expression <{expression}>"),
+                    ex.InnerException);
+            }

```

Comment:
I removed the `TargetInvocationException` and just wrapped its `InnerException`. I don't think the TIE provides any useful information.